library/src/tensile_host.cpp

```diff
@@ -0,0 +1,398 @@
+                                                        dt, dt, dt, dt,
+                                                        m, n, k, batchSize,
+                                                        ld_a, stride_a,
+                                                        ld_b, stride_b,
+                                                        ld_c, stride_c,
+                                                        ld_c, stride_c,
+                                                        *beta);
+
+    return problem;
+}
+
+
+// construct the gemm contraction problem
+template <typename T>
+Tensile::ContractionProblem create_gemm_contraction_problem (
+                                     rocblas_operation trans_a,
+                                     rocblas_operation trans_b,
+                                     unsigned long       m,
+                                     unsigned long       n,
+                                     unsigned long       k,
+                                     const T*          alpha,
+                                     const T*          A,
+                                     unsigned long       ld_a,
+                                     const T*          B,
+                                     unsigned long       ld_b,
+                                     const T*          beta,
+                                     T*                C,
+                                     unsigned long       ld_c)
+{
+
+    bool transposeA = false;
+    if (trans_a == rocblas_operation_conjugate_transpose)
+        transposeA = true;
+
+    bool transposeB = false;
+    if(trans_b == rocblas_operation_conjugate_transpose)
+        transposeB = true;    
+
+
+    Tensile::ContractionProblem::FreeIndex free;
+    Tensile::ContractionProblem::BoundIndex bound;
+
+    free.ca = free.da = 0;
+    free.cb = free.db = 1;
+
+    Tensile::TensorDescriptor a, b, c, d;
+
+    Tensile::DataType dt = tensile_datatype<T>();
+    if(transposeA)
+    {
+        a = Tensile::TensorDescriptor(dt, {k, m}, {1, ld_a});
+        free.a = 1;
+        bound.a = 0;
+    }
+    else
+    {
+        a = Tensile::TensorDescriptor(dt, {m, k}, {1, ld_a});
+        free.a = 0;
+        bound.a = 1;
+    }
+
+    if(transposeB)
+    {
+        b = Tensile::TensorDescriptor(dt, {n, k}, {1, ld_b});
+        free.b = 0;
+        bound.b = 1;
+    }
+    else
+    {
+        b = Tensile::TensorDescriptor(dt, {k, n}, {1, ld_b});
+        free.b = 1;
+        bound.b = 0;
+    }
+
+    Tensile::ContractionProblem::FreeIndices freeIndices{free};
+    Tensile::ContractionProblem::BatchIndices batchIndices;
+    Tensile::ContractionProblem::BoundIndices boundIndices{bound};
+
+    d = Tensile::TensorDescriptor(dt, {m, n}, {1, ld_c});
+
+    unsigned int batchCount = 1;
+
+    a.appendDim(batchCount);
+    b.appendDim(batchCount);
+    d.appendDim(batchCount);
+
+    batchIndices.push_back({2,2,2,2});
+
+    if(*beta != 0.0)
```

Comment:
`alpha` and `beta` can exist on either the device or host, depending on handle pointer mode.

If they exist on device, then `*alpha` and `*beta` cannot be dereferenced on host. They either need to wait and dereference on device, or `hipMemcpy` (synchronizing call) must be used to copy from device to host. This is best avoided if possible.